src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java

```diff
@@ -0,0 +1,158 @@
+package com.KacFlor.ShopSpring.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.KacFlor.ShopSpring.dao.UserRepository;
+import com.KacFlor.ShopSpring.model.Customer;
+import com.KacFlor.ShopSpring.model.Role;
+import com.KacFlor.ShopSpring.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willDoNothing;
+import static org.mockito.Mockito.*;
+import org.springframework.security.core.Authentication;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceTest{
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserService userService;
+
+    private User user1;
+    private User user2;
+
+    private User hAdmin;
+
+    @BeforeEach
+    public void setup() {
+        PasswordEncoder encoder = new BCryptPasswordEncoder();
+        user1 = new User("Test1", "encodedPassword1", new Customer(), Role.USER);
+        user2 = new User("Test2", "encodedPassword2", new Customer(), Role.USER);
+        hAdmin = new User("HAdmin",  encoder.encode("1234"), new Customer(), Role.ADMIN);
+    }
+
+    @DisplayName("JUnit test for getAllUsers method")
+    @Test
+    public void testGetAllUsers(){
+
+        User user3 = new User("Test3", "encodedPassword3", new Customer(), Role.USER);
+
+        given(userRepository.findAll()).willReturn(List.of(user1,user2,user3,hAdmin));
```

Comment:
I am not familiar with this test framework, this one here is a mock right?

Reply:
yes this is mock